### PR TITLE
Frontend login

### PR DESCRIPTION
**Pull request by MikolJiron**
Implemented frontend for the user to login/logout to the webpage.

### CODE REVIEW

File: portfolio/src/main/webapp/comments.js [hunk 1]
```diff
@@ -70,11 +72,46 @@ class Comments {
...
+   */
+  createLoginLogoutButton(isLoggedIn, loginLogoutURL) {
+    if (isLoggedIn) {
+      this.loginLogoutButton.innerHTML = `<a href=${loginLogoutURL}>Log Out</a>`;
+      this.loginLogoutStatusMessage.innerText = 'Welcome! You are logged in!';
+    } else {
+      this.loginLogoutButton.innerHTML = `<a href=${loginLogoutURL}>Log In</a>`;
+      this.loginLogoutStatusMessage.innerText = 'Access Denied. Please log in!';
+    }
+  }
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/comments.js:112]: Pretty sure this doesn't have anything to do with comments, right? This should be in its own class.

File: portfolio/src/main/webapp/authentication.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Comments} from './comments.js';
+
+class Authentication {
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/authentication.js:17]: JsDoc

File: portfolio/src/main/webapp/authentication.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   */
+  getLoginStatus() {
+    fetch('/authenticate-user')
+      .then(this.checkFetchError)
+      .then((loginStatus) => {
+        // Show comments if you're logged in.
+        if (loginStatus.isLoggedIn) {
+          const comments = new Comments();
+          comments.getComments(10);
+        }
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/authentication.js:34]: Move this elsewhere. This doesn't relate to authentication in any way. The easiest thing to do here would probably be to return the promise so you can chain another promise to do the comments stuff.

File: portfolio/src/main/webapp/authentication.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+   * Returns an error if the response status is not between 200 and 299, i.e not OK.
+   * @param {*} response - The HTTP response received from the servlet.
+   * @return {*} - If an error is not thrown, returns the json response.
+   */
+  checkFetchError(response) {
+    if (response.ok) {
+      return response.json();
+    } else {
+      throw Error(`${response.statusText}. Status: ${response.status}`);
+    }
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/authentication.js:69]: I'm not sure if you need a rebase or this is separate from the code I just reviewed, but please make the necessary changes here. In the future, please apply feedback from comments across all your PRs so we're not reviewing the same mistakes over and over.

File: portfolio/src/main/webapp/index.js [hunk 5]
```diff
@@ -21,17 +22,19 @@ import {Utils} from './utils.js';
...
   window.tab = tab;
   window.slideShow = slideShow;
   window.comments = comments;
-
+  window.authentication = authentication;
+  
   // Get the comments from the server and populate them into the 
   // comments-container.
-  comments.getComments(/** commentsLimit= */ 10);
+  authentication.getLoginStatus();
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/index.js:37]: This is where you would chain the comments logic.

File: portfolio/src/main/webapp/authentication.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  /**
+   * Create login-logout button depending on whether the user is logged in or not.
+   * @param {boolean} isLoggedIn - Is the user logged in? Yes or No?
+   * @param {string} loginLogoutURL - The URL the user will be
+   *   redirected once they click on the button.
+   */
+  createLoginLogoutButton(isLoggedIn, loginLogoutURL) {
+    if (isLoggedIn) {
+      this.loginLogoutButton.innerHTML = `<a href=${loginLogoutURL}>Log Out</a>`;
```
Comment (hzimz) [inline portfolio/src/main/webapp/authentication.js:51]: Move HTML outside of js using templates: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

Comment (hzimz) [reply portfolio/src/main/webapp/authentication.js:51]: Sorry, accidentally approved. Please address this comment before submitting

File: portfolio/src/main/webapp/authentication.js (hunk 4 above)
Comment (MikolJiron) [reply portfolio/src/main/webapp/authentication.js:69]: Just rebased it! So in my catch() statement I'll have a description of the error and where it occurred.

**Pull request merged**